resources/validations/src/test.sch

```diff
@@ -0,0 +1,127 @@
+
+    <sch:pattern>
+
+        <sch:rule
+            context="/">
+            <sch:p>Show distinct names used in props</sch:p>
+            <sch:report
+                role="INFO"
+                test="$INFO">Distinct prop names: <sch:value-of
+                    select="string-join(distinct-values(//oscal:prop/@name), ' ')"/></sch:report>
+        </sch:rule>
+
+        <sch:rule
+            context="oscal:*[starts-with(@href, '#')]">
+            <sch:p>Ensure all intra-document @hrefs have an intra-document target</sch:p>
+            <sch:let
+                name="ref"
+                value="substring-after(@href, '#')"/>
+            <sch:assert
+                role="ERROR"
+                test="$ref = //@uuid"><sch:value-of
+                    select="@rel"/>
+                <sch:name/> has invalid intra-document href <sch:value-of
+                    select="@href"/></sch:assert>
+            <sch:report
+                role="INFO"
+                test="$INFO and $ref = //@uuid"><sch:value-of
+                    select="@rel"/>
+                <sch:name/> has valid intra-document href to <sch:value-of
+                    select="local-name(//oscal:*[@uuid = $ref])"/>
+                <sch:value-of
+                    select="
+                        if (//oscal:*[@uuid = $ref]/oscal:title) then
+                            concat('&quot;', //oscal:*[@uuid = $ref]/oscal:title, '&quot;')
+                        else
+                            $ref"/></sch:report>
+        </sch:rule>
+
+        <sch:rule
+            context="oscal:back-matter">
+            <sch:p>It seems unlikely that most if not all SSP resources require a FedRAMP-specific prop.</sch:p>
+            <sch:p>The current fedramp-automation content appears to have presumed that NIST would supply (and be normative for) RMF prop types (NIST
+                has/is not) as opposed to FedRAMP prop types.</sch:p>
+            <sch:assert
+                test="oscal:resource[prop[@ns = 'https://fedramp.gov/ns/oscal']]">A FedRAMP SSP MUST include FedRAMP-specific attachments as
+                back-matter resources</sch:assert>
+        </sch:rule>
+
+        <sch:rule
+            context="oscal:back-matter/oscal:resource">
+            <!-- the following assertion recapitulates the XML Schema constraint -->
+            <sch:assert
+                role="ERROR"
+                test="@uuid">A &lt;<sch:name/>&gt; element MUST have a uuid attribute </sch:assert>
+            <sch:assert
+                role="ERROR"
+                test="oscal:prop[@ns = 'https://fedramp.gov/ns/oscal' and @name = 'type']">&lt;<sch:name/> uuid="<sch:value-of
+                    select="@uuid"/>"&gt; MUST have a &lt;prop&gt; element in the FedRAMP namespace</sch:assert>
+            <sch:assert
+                role="WARNING"
+                test="oscal:title">&lt;<sch:name/> uuid="<sch:value-of
+                    select="@uuid"/>"&gt; SHOULD have a title</sch:assert>
```

Comment:
How does expanding this `<sch:value-of select="@uuid" />` inside an attribute value work?

Reply:
The mixed content of the `<sch:assert>` begins just after `<sch:assert role="WARNING" test="oscal:title">`. The assertion test is just `oscal:prop[@ns = 'https://fedramp.gov/ns/oscal' and @name = 'type']`.

The content: `<&lt;<sch:name/> uuid="<sch:value-of select="@uuid"/>"&gt; MUST have..`

will be rendered (if the assertion fails) like

`<resource uuid="d45612a9-cf25-4ef6-b2dd-69e38ba2967a"> MUST have a  <prop> element in the FedRAMP namespace`.